src/thing_objects.c

```diff
@@ -1750,7 +1749,110 @@ TngUpdateRet object_update_object_scale(struct Thing *objtng)
 
 TngUpdateRet object_update_power_sight(struct Thing *objtng)
 {
-    return _DK_object_update_power_sight(objtng);
+    int result; // eax
+
+    objtng->health = 2;
+
+    static const long effkind[] = {43, 86, 87, 88};
+    struct Dungeon * dungeon = get_dungeon(objtng->owner);
+
+    if ( !S3DEmitterIsPlayingSample(objtng->snd_emitter_id, 51, 0) ) {
+        thing_play_sample(objtng, 51, 0x64u, -1, 3u, 1u, 3, 256);
+    }
+
+    int sight_casted_splevel = dungeon->sight_casted_splevel;
+
+    const struct MagicStats *pwrdynst = get_power_dynamic_stats(PwrK_SIGHT);
+    int max_time_active = pwrdynst->strength[sight_casted_splevel];
+
+    if ( game.play_gameturn - objtng->creation_turn >= max_time_active
+        && game.play_gameturn - dungeon->sight_casted_gameturn < max_time_active )
+    {
+        int time_active = game.play_gameturn - dungeon->sight_casted_gameturn;
+        if ( game.play_gameturn - dungeon->sight_casted_gameturn >= 0 )
+        {
+            if ( max_time_active / 16 < time_active )
+                time_active = max_time_active / 16;
+        }
+        else
+        {
+            time_active = 0;
+        }
+        const int v32 = (max_time_active / 16) / power_sight_close_instance_time[sight_casted_splevel];
```

Comment:
Are you able to name v32?

Reply:
It is used on the next line. I can get rid of it but that would make that line even longer. I'm not really sure what to call it... the calculations are a bit weird.

Reply:
I think it's weird too yes, with de divide in the middle and no brackets. I think it might be the different phases, where it is smaller at first and bigger when it's older, but I don't know a good name either.